src/runtime/mod.rs

```diff
@@ -124,10 +118,16 @@ impl Runtime {
     }
 }
 
-struct RuntimeContextGuard;
-
-impl Drop for RuntimeContextGuard {
+impl Drop for Runtime {
     fn drop(&mut self) {
+        // drop tasks
+        unsafe {
+            ManuallyDrop::drop(&mut self.local);
+            ManuallyDrop::drop(&mut self.rt);
+        }
+
+        // once tasks are dropped, we can unset the driver
+        // this will block until all completions are received
```

Comment:
Interesting. This shows the unset happens after local tasks and rt are dropped. Well, if that works, it seems very clean. The order is surprising given the order of creation in new, but I guess that's the point.